src/stp/Role.cpp

```diff
@@ -16,20 +16,24 @@ Status Role::update(StpInfo const& info) noexcept {
         return Status::Success;
     }
 
-    // Reset the role
-    if (robotTactics.current_num() != 0 && shouldRoleReset(info)) {
-        RTT_INFO("State Machine reset for current role for ID = ", info.getRobot()->get()->getId())
-        // TODO: messy reset, do it in the state machine
-        robotTactics.skip_n(-robotTactics.current_num());
-    }
-
     // Update tactic info
     auto tacticInfo = calculateInfoForTactic(info);
 
+    // Reset the role
+    if (robotTactics.current_num() != 0 && shouldRoleReset(tacticInfo)) {
+        RTT_INFO("State Machine reset for current role for ID = ", tacticInfo.getRobot()->get()->getId())
+        // Reset all the Tactics state machines
+        for (auto& tactic : robotTactics) {
+            tactic->getSkills().reset();
```

Comment:
Insted of resetting the actual state machine of each tactic, I would rather have the base tactic contain a reset method that resets the state machine of the tactic. It's cleaner and more maintainable.

```suggestion
            tactic->.reset();
```